File: backend/app/schemas/patient.py
from pydantic import BaseModel


class PatientListItem(BaseModel):
    id: str
    empi_id: str
    first_name: str
    last_name: str
    date_of_birth: str
    gender: str
    phone: str | None = None
    pathway_status: str | None = None
    pathway_name: str | None = None
    care_gaps: list[str] | None = None
    last_contact_date: str | None = None
    assigned_to: str | None = None
    pcp_name: str | None = None
    insurance_plan: str | None = None
    active_medications: list[dict] | None = None


class PatientDetail(PatientListItem):
    email: str | None = None
    cpf: str | None = None
    address: dict | None = None
    preferred_language: str = "pt"
    preferred_channel: str = "whatsapp"
    allergies: list[str] | None = None
    sdoh_flags: dict | None = None
    review_due_date: str | None = None


class PatientLabRecord(BaseModel):
    id: str
    test_type: str
    value: float
    unit: str
    source_system: str | None = None
    recorded_at: str


class PatientDiagnosisRecord(BaseModel):
    id: str
    icd10_code: str
    description: str
    diagnosed_at: str | None = None
    is_active: bool


class PatientListResponse(BaseModel):
    items: list[PatientListItem]
    total: int
    page: int
    page_size: int
    pages: int

File: src/services/types/patient.ts
export interface PatientListItem {
  id: string;
  empi_id: string;
  first_name: string;
  last_name: string;
  date_of_birth: string;
  gender: string;
  phone: string | null;
  pathway_status: string | null;
  pathway_name: string | null;
  care_gaps: string[] | null;
  last_contact_date: string | null;
  assigned_to: string | null;
  pcp_name: string | null;
  insurance_plan: string | null;
  active_medications: Array<{
    name: string;
    dose: string;
    frequency: string;
    pdc_90day: number;
  }> | null;
}

export interface PatientDetail extends PatientListItem {
  email: string | null;
  cpf: string | null;
  address: Record<string, string> | null;
  preferred_language: string;
  preferred_channel: string;
  allergies: string[] | null;
  sdoh_flags: Record<string, boolean | number> | null;
  review_due_date: string | null;
}

export interface PatientLabRecord {
  id: string;
  test_type: string;
  value: number;
  unit: string;
  source_system: string | null;
  recorded_at: string;
}

export interface PatientDiagnosisRecord {
  id: string;
  icd10_code: string;
  description: string;
  diagnosed_at: string | null;
  is_active: boolean;
}

export interface PatientListResponse {
  items: PatientListItem[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}
